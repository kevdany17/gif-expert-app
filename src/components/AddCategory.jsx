import { useState } from 'react';

export const AddCategory = ({ onNewValue }) => {
  const [value, setValue] = useState('One Piece');

  const onInputChange = ({ target }) => {
    setValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (value.trim().length <= 1) return;
    onNewValue(value.trim());
    setValue('');
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={value}
        onChange={onInputChange}
      />
    </form>
  );
};
