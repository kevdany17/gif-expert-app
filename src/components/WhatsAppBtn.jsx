import PropTypes from 'prop-types';
import whatsAppImage from '../../assets/social.png';

export const WhatsAppBtn = ({ url }) => {
  return (
    <a href={`https://api.whatsapp.com/send?text=${url}`} target="_blank">
      <img src={whatsAppImage} />
    </a>
  );
};

WhatsAppBtn.propTypes = {
  url: PropTypes.string.isRequired,
};
