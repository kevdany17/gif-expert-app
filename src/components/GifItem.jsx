import PropTypes from 'prop-types';
import { WhatsAppBtn } from './WhatsAppBtn';

export const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
      <WhatsAppBtn url={url} />
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
