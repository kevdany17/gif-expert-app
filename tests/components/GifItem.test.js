import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Test on GifItem', () => {
  const title = 'One Punch';
  const url = 'https://domain.com/image.jpg';

  test('should have snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should show image with url and alt', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    const { src, alt } = screen.getAllByRole('img')[0]; // the first element (show more images)
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should show title on component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
