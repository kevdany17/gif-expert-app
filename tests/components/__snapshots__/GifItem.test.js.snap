// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifItem should have snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://domain.com/image.jpg"
    />
    <p>
      One Punch
    </p>
    <a
      href="https://api.whatsapp.com/send?text=https://domain.com/image.jpg"
      target="_blank"
    >
      <img
        src="./assets/social.png"
      />
    </a>
  </div>
</div>
`;
